app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetControllerTest.kt

```diff
@@ -862,4 +862,62 @@ class BloodPressureEntrySheetControllerTest {
     verify(sheet).showInvalidDateError()
     verifyNoMoreInteractions(sheet)
   }
+
+  @Test
+  fun `when the data entry sheet changes date and back button is pressed, then update date button in BP entry`() {
+    val systolic = 120.toString()
+    val diastolic = 110.toString()
+    val localDate = LocalDate.of(2016, 5, 10)
+
+    whenever(bpValidator.validate(systolic, diastolic)).thenReturn(Success(systolic.toInt(), diastolic.toInt()))
+    whenever(dateValidator.validate2(any(), any())).thenReturn(Valid(localDate)) // TODO Use actual values
+
+    with(uiEvents) {
+      onNext(BloodPressureEntrySheetCreated(OpenAs.New(patientUuid)))
+      onNext(BloodPressureScreenChanged(BP_ENTRY))
+      onNext(BloodPressureSystolicTextChanged(systolic))
+      onNext(BloodPressureDiastolicTextChanged(diastolic))
+      onNext(BloodPressureDateClicked)
+      onNext(BloodPressureScreenChanged(DATE_ENTRY))
+      onNext(BloodPressureDayChanged("10"))
+      onNext(BloodPressureMonthChanged("5"))
+      onNext(BloodPressureYearChanged("16"))
+
+      reset(sheet)
+      onNext(BloodPressureBackPressed)
+    }
+
+    verify(sheet).showDate(localDate)
+    verify(sheet).showBpEntryScreen()
+    verifyNoMoreInteractions(sheet)
+  }
+
+  @Test
+  fun `when the data entry sheet changes date and show BP button is pressed, then update date button in BP entry`() {
+    val systolic = 120.toString()
+    val diastolic = 110.toString()
+    val localDate = LocalDate.of(2016, 5, 10)
+
+    whenever(bpValidator.validate(systolic, diastolic)).thenReturn(Success(systolic.toInt(), diastolic.toInt()))
+    whenever(dateValidator.validate2(any(), any())).thenReturn(Valid(localDate)) // TODO Use actual values
```

Comment:
What is this TODO for?

Reply:
We are supposed to reach a point where the use of argument matchers should be justified. This is one step towards that goal.